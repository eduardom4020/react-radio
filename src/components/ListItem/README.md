# ListItem
>This component is a View that renders two text sparsed with an inferior border.

### Props
These are the props accepted by the component, pay attention to the description, inputs and behaviour:

---

##### color *(string)*
*The background color of the wrapper. It can be colored passing color values accepted by CSS. By default it value is white.*
**example**: 
```html
<ListItem color='#a3acbe' />
```

---

##### borderColor *(string)*
*The border color of the wrapper. It can be colored passing color values accepted by CSS. By default it value is #a3acbe.*
**example**: 
```html
<ListItem borderColor='grey' />
```

---

##### borderVisible *(boolean)*
*If false, hides the bottom border of this component.*
**example**: 
```html
<ListItem borderVisible={true} />
```

---