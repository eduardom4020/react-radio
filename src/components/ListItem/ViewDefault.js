import styled from 'styled-components';
import { View } from 'react-native';

const ListItemViewDefault = styled(View)`
    width: 100%;
    height: 64px;
    background-color: ${props => props.color || 'white'};
    flex-direction: row;
    justify-content: space-between;
    align-items: center;
    border-style: solid;
    border-bottom-width: ${props => props.borderVisible === false ? 0 : 1}px;
    border-bottom-color: #a3acbe;
`;

export default ListItemViewDefault;