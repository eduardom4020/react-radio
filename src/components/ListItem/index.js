import View from './ViewDefault';
export default View;