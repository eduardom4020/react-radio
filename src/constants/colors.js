export const BLACK_LIKE = '#20202b';
export const DARK_GREY = '#2e2e3a';
export const MEDIUM_GREY = '#a3acbe';
export const MEDIUM_ORANGE = '#eeae61';
export const METALIC_BLACK = '#191313';