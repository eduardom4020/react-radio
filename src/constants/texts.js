export const APP_TITLE = 'STATIONS';