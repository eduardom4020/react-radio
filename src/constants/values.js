export const DEFAULT_IMAGE_BT_SIZE = 192;
export const DEFAULT_ICON_BT_SIZE = 24;
export const DEFAULT_RADIO_PLAYING = 'CURRENTLY PLAYING'