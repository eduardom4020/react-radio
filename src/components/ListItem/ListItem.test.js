import React from 'react';
import { Text, View } from 'react-native';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import 'jest-styled-components';

import { DARK_GREY } from '~/src/constants/colors';
import ListItem from '~/src/components/ListItem';

describe('ListItem', () => {
    test('Render?', () => {
        const component = shallow(
            <ListItem />
        );

        expect(toJson(
            component, {
                noKey: false,
                mode: 'deep'
            }
        )).toMatchSnapshot();
    });

    test('Render with custom color?', () => {
        const component = shallow(
            <ListItem color={DARK_GREY} />
        );

        expect(toJson(
            component, {
                noKey: false,
                mode: 'deep'
            }
        )).toMatchSnapshot();
    });

    test('Render with custom border color?', () => {
        const component = shallow(
            <ListItem borderColor={DARK_GREY} />
        );

        expect(toJson(
            component, {
                noKey: false,
                mode: 'deep'
            }
        )).toMatchSnapshot();
    });
    
    test('Render with children?', () => {
        const component = shallow(
            <ListItem >
                <View><Text>Test1</Text></View>
                <Text>Test2</Text>
            </ListItem>
        );

        expect(toJson(
            component, {
                noKey: false,
                mode: 'deep'
            }
        )).toMatchSnapshot();
    });
});