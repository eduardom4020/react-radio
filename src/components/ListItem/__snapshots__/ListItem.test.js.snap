// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem Render with children? 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "
    width: 100%;
    min-height: 82px;
    background-color: ",
          [Function],
          ";
    flex-direction: row;
    justify-content: space-between;
    align-items: center;
    flex-wrap: wrap;
    border-style: solid;
    border-bottom-width: ",
          [Function],
          "px;
    border-bottom-color: ",
          [Function],
          ";
",
        ],
      },
      "displayName": "Styled(View)",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": [Function],
      "toString": [Function],
      "usesTheme": false,
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
>
  <View>
    <Text>
      Test1
    </Text>
  </View>
  <Text>
    Test2
  </Text>
</StyledComponent>
`;

exports[`ListItem Render with custom border color? 1`] = `
<StyledComponent
  borderColor="#2e2e3a"
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "
    width: 100%;
    min-height: 82px;
    background-color: ",
          [Function],
          ";
    flex-direction: row;
    justify-content: space-between;
    align-items: center;
    flex-wrap: wrap;
    border-style: solid;
    border-bottom-width: ",
          [Function],
          "px;
    border-bottom-color: ",
          [Function],
          ";
",
        ],
      },
      "displayName": "Styled(View)",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": [Function],
      "toString": [Function],
      "usesTheme": false,
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;

exports[`ListItem Render with custom color? 1`] = `
<StyledComponent
  color="#2e2e3a"
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "
    width: 100%;
    min-height: 82px;
    background-color: ",
          [Function],
          ";
    flex-direction: row;
    justify-content: space-between;
    align-items: center;
    flex-wrap: wrap;
    border-style: solid;
    border-bottom-width: ",
          [Function],
          "px;
    border-bottom-color: ",
          [Function],
          ";
",
        ],
      },
      "displayName": "Styled(View)",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": [Function],
      "toString": [Function],
      "usesTheme": false,
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;

exports[`ListItem Render? 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "
    width: 100%;
    min-height: 82px;
    background-color: ",
          [Function],
          ";
    flex-direction: row;
    justify-content: space-between;
    align-items: center;
    flex-wrap: wrap;
    border-style: solid;
    border-bottom-width: ",
          [Function],
          "px;
    border-bottom-color: ",
          [Function],
          ";
",
        ],
      },
      "displayName": "Styled(View)",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": [Function],
      "toString": [Function],
      "usesTheme": false,
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;
